tls/s2n_connection.c

```diff
@@ -534,6 +534,9 @@ int s2n_connection_wipe(struct s2n_connection *conn)
     GUARD(s2n_connection_init_hashes(conn));
     GUARD(s2n_connection_init_hmacs(conn));
 
+    /* Require all handshakes hashes. This set can be reduced as the handshake progresses. */
+    GUARD(s2n_handshake_require_all_hashes(conn));
```

Comment:
You could modify `s2n_conn_update_required_handshake_hashes` to be used on `s2n_connection_wipe` as well, keeping a single point of entry for updating the required handshake hashes. Any thoughts?

Reply:
I'm inclined to leave it as-is to reduce the complexity of s2n_conn_update_required_handshake_hashes. It's simple because we assume it can only be called during 2 handshake messages. Exposing s2n_handshake_require_all_hashes does not introduce any extra cognitive burden. If anything it is more intuitive.